gk/main.c

```diff
@@ -1996,13 +1997,21 @@ static int
 gk_stage1(void *arg)
 {
 	struct gk_config *gk_conf = arg;
-	int ret, i;
+	struct gatekeeper_if *front_if = &gk_conf->net->front;
+	int ret, i, TABLE_SIZE = front_if->num_rx_queues + 1;
 
 	gk_conf->instances = rte_calloc(__func__, gk_conf->num_lcores,
 		sizeof(struct gk_instance), 0);
-	if (gk_conf->instances == NULL)
+
+	gk_conf->queue_id_to_instance = rte_malloc(__func__,
+		TABLE_SIZE*sizeof(int), 0);
+
+	if (gk_conf->instances == NULL ||
+		gk_conf->queue_id_to_instance == NULL)
```

Comment:
Style: break this `if` statement into two `if` statements, one after each memory allocation above.